docs/adr/0005-feature-flags.md

```diff
@@ -0,0 +1,149 @@
+* CI/CD: simpler merges into the main branch
+* Environment-dependent configurations
+* Experimentations: A/B testing and different experiences for different users based on context (user, device, geolocation data etc)
+
+## Decision Drivers <!-- optional -->
+
+When applying feature flagging, we often have a choice between making that toggle on the frontend or the backend. Here are a few factors that we should keep in mind while making that decision
+
+### UI Performance
+By moving flagging decisions to the server side, we are reducing code bloat on the frontend, gaining performance that's immediately noticeable by the user. If we decide to move Monitor frontend to SPA in the future, Single-page applications are already making a API call to render the data needed for the UI. With the same payload, we can make a call to a feature-flag service, so one network call can contain all feature-flag configs with the server-side data.
+
+### Config Cache Lag 
+There are two general approaches:
+
+1. An app can proactively request flagging decisions based on runtime-context
+
+> This approach consumes a lot more bandwidth (every context changes = new feature flag decision being made by the backend). It's a lot more flexible and prevents cache lag
+2. An app can request general config and use client-side router
+
+> With this approach, we essentially cache the config. In the case of a kill switch, the frontend may not be able to react quickly to the changes from the backend.
+
+### Security
+One key consideration when deciding where to implement a feature flag is security. If a feature is highly sensitive or critical to the security of the application, it may be safer to implement the toggle on the backend. This can help ensure that the feature is properly secured and that user data is protected.
+
+### Sync Complexity
+Implementing feature flagging on the server side reduces the complexity of synchronizing feature toggle logic between the frontend and backend, as there is a single central location to manage the toggle settings. This simplifies development and ensures that the system runs smoothly, as dependencies flow in one direction.
+## Maintenance & Best Practices
+### Naming Flags
+Defining a pattern or naming convention is generally a good practice for keeping things organized. Engineers can quickly identify from the name what the feature is about and be able to recognize what areas of the application for which the feature is being used. I'll list a few useful examples of naming structures and demonstrate the practical benefits.
+
+A good naming **convention** I found is as follows:
+```
+section_purpose_service
+```
+
+The feature name example that follows has three parts:
+1. section (optional): we present the name of the section of the app the feature is gating (free/premium). 
```

Comment:
Could you clarify whether this is an exhaustive list? I'm always struggling figuring out what a "section" is, e.g. is it the settings page? Or a specific section of the settings page, such as Premium settings? What if a feature is available to both premium and free subscribers?

Alternatively, I'd also be fine with just skipping `section`, to avoid [smurf naming](https://blog.codinghorror.com/new-programming-jargon/#21).

Reply:
When naming something, I always think about future proofing, and namespaces. To namespace, I think we always want to think from large concepts to smaller concepts.

I don't know if "section" is the best way to word to describe what I was thinking there.. but I was thinking more about the different "versions" an app can exist in: free version, premium version, admin version (potentially)